proposals/NNNN-init-accessors.md

```diff
@@ -0,0 +1,362 @@
+  }
+}
+```
+
+An assignment to a stored property before all of `self` is initialized will initialize that stored property. When all of the stored properties listed in the `initializes:` clause of a computed property with an `init` accessor have been initialized, that computed property is virtually initialized:
+
+```swift
+struct S {
+  var x1: Int
+  var x2: Int
+  var x3: Int
+  var computed: Int {
+    init(newValue) initializes(x1, x2) { ... }
+  }
+
+  init() {
+    self.x1 = 1 // initializes 'x1'; neither 'x2' or 'computed' is initialized
+    self.x2 = 1 // initializes 'x2' and 'computed'
+    self.x3 = 1 // initializes 'x3'; 'self' is now fully initialized
+  }
+}
+```
+
+An assignment to a computed property where at least one of the stored properties listed in `initializes:` is initialized, but `self` is not initialized, is an error. This prevents double-initialization of the underlying stored properties:
+
+```swift
+struct S {
+  var x: Int
+  var y: Int
+  var point: (Int, Int) {
+    init(newValue) initializes(x, y) {
+	    (self.x, self.y) = newValue
+    }
+    get { (x, y) }
+    set { (x, y) = newValue }
+  }
+
+  init(x: Int, y: Int) {
+    self.x = x // Only initializes 'x'
+    self.point = (x, y) // error: neither the `init` accessor nor the setter can be called here
+  }
+}
+```
+
+### Memberwise initializers
+
+If a struct does not declare its own initializers, it receives an implicit memberwise initializer based on the stored properties of the struct, because the storage is what needs to be initialized. Because `init` provide a preferred mechanism for initializing storage, the memberwise initializer parameter list will include any computed properties that subsume the initialization of stored properties instead of parameters for those stored properties.
+
+```swift
+struct S {
+  var _x: Int
+  var x: Int {
+    init(newValue) initializes(_x) {
+      _x = newValue
+    }
+
+    get { _x }
+    set { _x = newValue }
+  }
+
+  var y: Int  
+}
+
+S(x: 10, y: 100)
+```
+
+The above struct `S` receives a synthesized initializer:
+
+```swift
+init(x: Int, y: Int) {
+  self.x = x
+  self.y = y
+}
+```
+
+A memberwise initializer cannot be synthesized if a stored property that is an `accesses` effect of a computed property is ordered after that computed property in the source code:
+
+```swift
+struct S {
+  var _x: Int
+  var x: Int {
+    init(newValue) initializes(_x) accesses(y) {
+      _x = newValue
+    }
+
+    get { _x }
+    set { _x = newValue }
+  }
+
+  var y: Int 
+}
+```
+
+The above struct would receive the following memberwise initializer, which is invalid so an error is emitted:
+
+```swift
+init(x: Int, y: Int) {
+  self.x = x // error
+  self.y = y
+}
+```
+
+Note that macro-expanded declarations are ordered after the attached-to declaration for peer macros, and at the end of the member list as written for member macros.
```

Comment:
This line feels kind of like a non-sequitur and would probably benefit from an example. I'm not quite sure what it's saying or why it's germane in the discussion of `init` accessors.